consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -0,0 +1,765 @@
+            mock_consensus_state_store,
+            mock_poet_key_state_store,
+            mock_signup_info,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ Test verifies that PoET Block Publisher fails if
+        a block is claimed by an unknown validator (the validator is not listed
+        in the validator registry)
+        """
+
+        # create a mock_validator_registry_view that throws KeyError
+        mock_validator_registry_view.return_value.get_validator_info. \
+            side_effect = KeyError('Non-existent validator')
+
+        # create a mock_wait_certificate that does nothing in check_valid
+        mock_wait_certificate = mock.Mock()
+        mock_wait_certificate.check_valid.return_value = None
+
+        mock_utils.deserialize_wait_certificate.return_value = \
+            mock_wait_certificate
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_consensus_state.consensus_state_for_block_id.return_value = \
+            mock_state
+
+        mock_consensus_state_store.return_value.__getitem__.return_value = \
+            mock_consensus_state
+
+        # create mock_signup_info
+        mock_signup_info.create_signup_info.return_value = \
+            mock.Mock(
+                poet_public_key='poet public key',
+                proof_data='proof data',
+                anti_sybil_id='anti-sybil ID',
+                sealed_signup_data='sealed signup data')
+
+        # create mock_batch_publisher
+        mock_batch_publisher = mock.Mock(
+            identity_signing_key=signing.generate_privkey())
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_block_header with the following fields
+        mock_block = mock.Mock(identifier='0123456789abcdefedcba9876543210')
+        mock_block.header.signer_pubkey = '90834587139405781349807435098745'
+        mock_block.header.previous_block_id = '2'
+        mock_block.header.block_num = 1
+        mock_block.header.state_root_hash = '6'
+        mock_block.header.batch_ids = '4'
+
+        # check test
+        block_publisher = \
+            poet_block_publisher.PoetBlockPublisher(
+                block_cache=mock_block_cache,
+                state_view_factory=mock_state_view_factory,
+                batch_publisher=mock_batch_publisher,
+                data_dir=self._temp_dir,
+                validator_id='validator_deadbeef')
+
+        self.assertFalse(
+            block_publisher.initialize_block(
+                block_header=mock_block.header))
```

Comment:
I think you want to make a second check (like checking the LOGGER) to verify that things are really failing for the reason you think they are.  Currently, I cannot think of a better way to ensure this than using the LOGGER debug/error message strings.